From pavel@ucw.cz  Fri Aug  7 13:41:27 2009
From: Pavel Machek <pavel@ucw.cz>
Date: Wed, 5 Aug 2009 00:00:10 +0200
Subject: Staging: dream: synaptics touchscreen for dream: documentation
To: Greg KH <greg@kroah.com>
Cc: <arve@android.com>
Message-ID: <20090804220009.GA2077@elf.ucw.cz>
Content-Disposition: inline



This adds pointer to hardware documentation, and adds code comment
from Arve.

Signed-off-by: Pavel Machek <pavel@ucw.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/dream/synaptics_i2c_rmi.c |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

--- a/drivers/staging/dream/synaptics_i2c_rmi.c
+++ b/drivers/staging/dream/synaptics_i2c_rmi.c
@@ -13,6 +13,7 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
  *
+ * http://www.synaptics.com/sites/default/files/511_000099_01F.pdf
  */
 
 #include <linux/module.h>
@@ -87,6 +88,22 @@ static int synaptics_init_panel(struct s
 
 static void decode_report(struct synaptics_ts_data *ts, u8 *buf)
 {
+/*
+ * This sensor sends two 6-byte absolute finger reports, an optional
+ * 2-byte relative report followed by a status byte. This function
+ * reads the two finger reports and transforms the coordinates
+ * according the platform data so they can be aligned with the lcd
+ * behind the touchscreen. Typically we flip the y-axis since the
+ * sensor uses the bottom left corner as the origin, but if the sensor
+ * is mounted upside down the platform data will request that the
+ * x-axis should be flipped instead. The snap to inactive edge border
+ * are used to allow tapping the edges of the screen on the G1. The
+ * active area of the touchscreen is smaller than the lcd. When the
+ * finger gets close the edge of the screen we snap it to the
+ * edge. This allows ui elements at the edge of the screen to be hit,
+ * and it prevents hitting ui elements that are not at the edge of the
+ * screen when the finger is touching the edge.
+ */
 	int pos[2][2];
 	int f, a;
 	int base = 2;
